Tag functions to core IR in native_functions.yaml

SS-JIA opened the pull request:
Summary:
Based on operator review meetings, tag appropriate functions as part of the Core IR.

[Operator Review Tracking Sheet](https://docs.google.com/spreadsheets/d/1u9jQ-uGlKu-fe9nLy-jS2AIPtpE8sGTmELOFYgKOhXU/edit#gid=0)

Test Plan: Use N3940835 to load the YAML and check updated core op list.

Reviewed By: mergennachin, kimishpatel, SherlockNoMad

Differential Revision: D47673670

SherlockNoMad commented:
FYI, you might also need to update test/expect/HasDecompTest.test_aten_core_operators.expect

see https://github.com/pytorch/pytorch/pull/105673 for ref.

jgong5 reviewed (commented):
It seems not all tagged ops are documented in the spec https://pytorch.org/docs/stable/ir.html. Will the doc be updated accordingly? And does the "core IR" contain ops in both ATen IR and Prims IR?

jgong5 commented:
cc @sujoysaraswati

SS-JIA commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1208,7 +1208,7 @@
   variants: function, method
   dispatch:
     CompositeExplicitAutograd: logical_xor
-  tags: pointwise
+  tags: [core, pointwise]
```
Yeah, so to clarify; that page is generated based on which ops are tagged as "core" from this YAML file. So once this is updated, it should be reflected in the page as well. The list of ops on the page is currently incomplete, and at the moment we are working on expanding the list. Hence this PR.

SS-JIA commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1919,6 +1919,7 @@
   structured_delegate: cumsum.out
   device_check: NoCheck   # TensorIterator
   variants: function, method
+  tags: core
```
See above comment.

SS-JIA committed:
Tag functions to core IR in native_functions.yaml (#105849)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105849

Based on operator review meetings, tag appropriate functions as part of the Core IR.

[Operator Review Tracking Sheet](https://docs.google.com/spreadsheets/d/1u9jQ-uGlKu-fe9nLy-jS2AIPtpE8sGTmELOFYgKOhXU/edit#gid=0)

Test Plan: Use N3940835 to load the YAML and check updated core op list.

Reviewed By: mergennachin, kimishpatel, SherlockNoMad

Differential Revision: D47673670

fbshipit-source-id: e3b1afff0318451ce75d403a96844b0aee7d93a4
